numpy/ma/core.pyi

```diff
@@ -455,7 +462,68 @@ class MaskedIterator:
 
 class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     __array_priority__: Any
-    def __new__(cls, data=..., mask=..., dtype=..., copy=..., subok=..., ndmin=..., fill_value=..., keep_mask=..., hard_mask=..., shrink=..., order=...): ...
+    @overload
+    def __new__(
+        cls,
+        data: _ArrayLike[_ScalarT],
+        mask: _ArrayLikeBool_co = nomask,
+        dtype: None = None,
+        copy: bool = False,
+        subok: bool = True,
+        ndmin: int = 0,
+        fill_value: _ScalarLike_co | None = None,
+        keep_mask: bool = True,
+        hard_mask: bool | None = None,
+        shrink: bool = True,
+        order: _OrderKACF | None = None,
+    ) -> _MaskedArray[_ScalarT]: ...
+    @overload
+    def __new__(
+        cls,
+        data: Any,
+        mask: _ArrayLikeBool_co,
+        dtype: _DTypeLike[_ScalarT],
+        copy: bool = False,
+        subok: bool = True,
+        ndmin: int = 0,
```

Comment:
this returns an explicit shape type, so we can't have `ndim` interfere here:

```suggestion
        ndmin: Literal[0] = 0,
```

Reply:
right, thanks - maybe fine to just type the dtype then, as is done for `array`?

https://github.com/numpy/numpy/blob/6eeaae25ddfacf586ee02ddc600618127fa79cfd/numpy/_core/multiarray.pyi#L474-L484

Reply:
Yea I think that's wise. Shape-typing could be nice to have here, but it'll probably  complicate things quite a bit.